lib/mix/tasks/erlang.check_version.ex

```diff
@@ -30,12 +30,18 @@ defmodule Mix.Tasks.Erlang.CheckVersion do
     |> Path.join()
     |> File.read()
     |> case do
-      {:ok, version} -> sanitize_version(String.trim(version))
-      _ -> nil
+      {:ok, version} ->
+        version
+        |> String.trim()
+        |> sanitize_version_without_patch()
+        |> sanitize_version_with_extra_digits()
```

Comment:
Can’t we just do "21.1.2" split with ".", Pad with "0" (or remove elements to have 3 elements) and join with "."?

21.1 would give 21.1.0
21.1.2.3.4 would give 21.1.2

Do not we really need regex for that?

Reply:
I like @simonprev proposition!

```elixir
version
|> String.trim()
|> String.split(".")
|> Enum.concat(~w(0 0)) 
|> Enum.slice(0, 3) 
|> Enum.join(".")
```